graph/src/data/graphql/validation.rs

```diff
@@ -60,3 +63,12 @@ pub fn get_object_type_directive(object_type: &ObjectType, name: Name) -> Option
         .iter()
         .find(|directive| directive.name == name)
 }
+
+/// Returns the value type for a GraphQL field type.
+pub fn get_valid_value_type(field_type: &Type) -> Result<ValueType, Error> {
```

Comment:
I'm not sure this has anything to do with validation, so I'd put it next to the `attribute_index_operations()` function for now. And I'd call it `inner_type_name` instead of `get_valid_value_type`.